### PR TITLE
Tests: expect test for ppx syntax

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by Drup** [comment]
You're missing the extra opam dependency.

I still feel ppx_expect has far too many dependencies for what it does.

**Comment by hhugo** [comment]
Tests are currently never run (sadly) so the extra dependency doesn't really matter here.
This PR relies on Toplevel_expect_tests and makes tests more robust (see https://github.com/ocsigen/js_of_ocaml/issues/523)

**Comment by Drup** [comment]
Yeah, I read it, and it looks good. I still think you should add the `{test}` dependencies (so that people can do `opam install -t --deps-only` to start doing dev on jsoo.

**Comment by hhugo** [comment]
agreed.

File: tests/ppx/main_deriving.ml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Type_conv = Ppx_type_conv.Type_conv
+
+let l = Ppx_deriving.derivers ()
+let _ = ListLabels.iter l ~f:(fun (deriver : Ppx_deriving.deriver) ->
+  Type_conv.add deriver.Ppx_deriving.name
```
Comment (hhugo) [inline tests/ppx/main_deriving.ml]: @diml: `type_conv` derivers can be used with `ppx_deriving` using `ppx_type_conv.deriving`.
What about `ppx_deriving` derivers used with `type_conv` ?

Comment (diml) [inline tests/ppx/main_deriving.ml]: The feature is implemented and released in our tree, it will be on github next time we push, so hopefully next week

**Pull request merged**